siliconcompiler/schema/namedschema.py

```
@@ -66,17 +66,22 @@ def _getdict_type(cls) -> str:
 
         raise NotImplementedError("Must be implemented by the child classes.")
 
+    def _getdict_meta(self) -> Dict[str, str]:
+        info = super()._getdict_meta()
+        info["name"] = self.name
+        return info
+
     @classmethod
-    def from_manifest(cls, name: str, filepath: str = None, cfg: Dict = None):
+    def from_manifest(cls, filepath: str = None, cfg: Dict = None, name: str = None):
```

Ruff (0.13.2)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)